### ISSUE TITLE
Game elements showing prematurely in story screen

### ISSUE THREAD

**Issue by Wedge009** [wesnoth/wesnoth#6922]
### Game and System Information

- **Version:** Current master (6701542f633990ec14f9302d9e998062ad13d749)
- **Downloaded from:** Self-compiled
- **Build info:** SCons and MSVC
- **OS:** Linux/X11 and Windows

### Description of the bug

When making the transition from story screen to the main game screen, the top-left menu buttons and the mini-map buttons show prematurely. This happens whether using the skip button in the bottom-right or stepping through the story screens one-by-one.

Illustration:
![1](https://user-images.githubusercontent.com/13679322/181865501-cb61a57c-38b5-4a78-aef2-a2830f26f24c.jpg)

### Steps to reproduce the behavior

1. I was just testing Northern Rebirth but I imagine this happens on all campaigns - start a campaign.
2. Step through story screens.
3. Observe the game elements showing prematurely in the transition from story to game.

### Expected behavior

Not a major issue, but ideally these elements shouldn't show until the story screen is gone.

**Comment by Wedge009** [comment]
@mesilliac I don't recall seeing this before, so maybe it's due to a fairly recent change.

**Comment by mesilliac** [comment]
It's probably to do with the way the theme elements are now rendered. I basically hacked it such that the old GUI1 buttons you see here are independent top-level elements. I'm not sure if there's something raising them prematurely or something is drawing them when it shouldn't. Or possibly they're being entirely recreated... there is a lot of dirty stuff going on with those buttons.

Hopefully it will get fixed as part of general cleanup. If not i'll have to find some way to hide them.

**Comment by max-torch** [comment]
I said this on Discord:

I noticed something that I can add
It is not limited to the story screen
Even in the load game screen
Basically, any time the a scenario is loaded, in any context, the UI buttons appear onscreen first.

**Comment by Wedge009** [comment]
Confirmed it's specific to bringing up the game screen, so it's in the transition from loading screen as well. As per above remarks, it's related to the work-arounds for elements that are still using GUI1.

**Comment by Wedge009** [comment]
git bisect shows this started in 0548f5a490f631b634ad8288755fedfce4e4f8ff, introduced in 1.17.7.

**Comment by Toranks** [comment]
![imagen](https://github.com/wesnoth/wesnoth/assets/40789364/f22e54ba-9ab1-42a9-83f0-6d2b87adbd69)

**Comment by cooljeanius** [comment]
> It's probably to do with the way the theme elements are now rendered. I basically hacked it such that the old GUI1 buttons you see here are independent top-level elements. I'm not sure if there's something raising them prematurely or something is drawing them when it shouldn't. Or possibly they're being entirely recreated... there is a lot of dirty stuff going on with those buttons.
> 
> Hopefully it will get fixed as part of general cleanup. If not i'll have to find some way to hide them.

@mesilliac any additional hints on where in the code to try looking?

**Comment by mesilliac** [comment]
I don't remember what the code path is to get into game. I actually fixed a very similar issue to this at one point. The buttons probably get drawn somewhere in display.cpp when it initializes the theme. Like `display::set_theme` maybe. It could just hide the stuff until it's ready to show it. The old theme elements have a method to hide them. IIRC they should be in `widgets/button.cpp` etc.

There might be a simpler solution, but worst case is the theme stuff can start off hidden with a function to unhide it all when ready.

The reason only the buttons are getting drawn is that they got added in as top-level drawables so they could be handled the same way as everything else. There might also be a solution with raising the story screen to the top of the draw stack so it covers them. But that might also hide the buttons underneath the main game display afterwards, i'm not sure.
